scripts/remove-idle-users.py

```diff
@@ -0,0 +1,111 @@
+    for user_id in user_ids:
+        # Ignore unclaimed users, if user_template is specified
+        if user_template and not claims_user_id(user_id, user_template, homeserver_domain):
+            continue
+        res = requests.post(
+            homeserver + "/_matrix/client/r0/rooms/" + urllib.quote(room_id) + "/kick?access_token=" + token,
+            data = json.dumps({
+                "reason": reason,
+                "user_id": user_id
+            })
+        )
+        res.raise_for_status()
+
+def claims_user_id(user_id, user_template, homeserver_domain):
+    # the server claims the given user ID if the ID matches the user ID template.
+    regex = template_to_regex(
+        user_template,
+        {
+            "$SERVER": homeserver_domain
+        },
+        {
+            "$NICK": "(.*)"
+        },
+        escapeRegExp(":" + homeserver_domain)
+    )
+    print("Matching %s to %s" % (regex, user_id))
+    return re.match(regex, user_id)
+
+def template_to_regex(template, literal_vars, regex_vars, suffix = ""):
+    # The 'template' is a literal string with some special variables which need
+    # to be find/replaced.
+    regex = template;
+    for k in literal_vars:
+        regex = re.sub(escapeRegExp(k), regex, literal_vars[k])
+
+    # at this point the template is still a literal string, so escape it before
+    # applying the regex vars.
+    regex = escapeRegExp(regex);
+    # apply regex vars
+
+    for k in regex_vars:
+        regex = re.sub(
+            # double escape, because we bluntly escaped the entire string before
+            # so our match is now escaped.
+            escapeRegExp(escapeRegExp(k)), regex, regex_vars[k]
+        )
+
+    return regex + suffix
+
+def escapeRegExp(s):
+    return re.escape(s);
+
+
+def main(token, alias, homeserver, homeserver_domain, since, user_template):
+    print("Removing idle users in %s" % alias)
+    room_id = get_room_id(homeserver, alias, token)
+    if not room_id:
+        raise Exception("Cannot resolve room alias to room_id")
+
+    kick_idlers(homeserver, homeserver_domain, room_id, token, since, user_template)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser("Remove idle users from a given Matrix room")
+    parser.add_argument("-t", "--token", help="The access token", required=True)
+    parser.add_argument("-a", "--alias", help="The alias of the room eg '#freenode_#matrix-dev:matrix.org'", required=True)
+    parser.add_argument("-u", "--homeserver", help="Base homeserver URL eg 'https://matrix.org'", required=True)
+    parser.add_argument("-d", "--domain", help="Domain of the homserver eg 'matrix.org'", required=True)
+    parser.add_argument("-s", "--since", type=int, help="Days since idle users have been offline for eg '30'", required=True)
+    parser.add_argument("-e", "--template", help="User template to determine whether a user should be kicked. E.g. @$SERVER_$NICK", required=True)
```

Comment:
I would just do prefix matching to save yourself the hassle from trying to work out what to correctly parse things as. It also makes it clearer to the end-user what they should put as a template value ("@freenode_" for example). Bear in mind people who run this may not know the IRC bridge config file syntax so `$SERVER` and `$NICK` may be foreign to them.